File: BubbleSortAlgorithm/C++/BubbleSort.cpp
#include <iostream>
#include <vector>

void bubbleSort(std :: vector<int>& arr) {
    int n = arr.size();
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                std :: swap(arr[j], arr[j+1]);
            }
        }
    }
}

int main() {
    std :: vector<int> arr = {12, 92, 65, 4, 27, 19, 88};

    bubbleSort(arr);

    printf("Sorted array: ");
    for (int i = 0; i < arr.size(); i++) {
        std :: cout <<arr[i] <<" ";
    }
    std :: cout << std :: endl;

    return 0;
}

File: BubbleSortAlgorithm/Go/BubbleSort.go
package main

import (
	"fmt"
)

/*
Bubble Sort Implimentation in Go

Time Complexity O(n^2)
*/
func main() {
	var arr = []int{24, 9, 15, 1, 4, 8, 6}
	fmt.Println("Before sort: ", arr)
	bubbleSort(arr)
	fmt.Println("After sort: ", arr)
}

func bubbleSort(arr []int) {
	for i := 0; i < len(arr); i++ {
		for j := 1; j < len(arr)-i; j++ {
			if arr[j-1] > arr[j] {
				swap(arr, j-1, j)
			}
		}
	}
}

func swap(arr []int, i, j int) {
	var temp = arr[i]
	arr[i] = arr[j]
	arr[j] = temp
 
